cdc panic on syncpoint event encoding

### What did you do?

panic: SyncPointEvent: invalid version

goroutine 2570 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc21af0dec0?})
        go.uber.org/zap@v1.27.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc074846f70, {0xc07484b600, 0x2, 0x2})
        go.uber.org/zap@v1.27.0/zapcore/entry.go:262 +0x24a
go.uber.org/zap.(*Logger).Panic(0x4ee7f80?, {0x5bc9a9c?, 0x7f721e608b50?}, {0xc07484b600, 0x2, 0x2})
        go.uber.org/zap@v1.27.0/logger.go:285 +0x4b
github.com/pingcap/log.Panic({0x5bc9a9c?, 0xcb13000000000000?}, {0xc07484b600?, 0x47ce49?, 0x14cc?})
        github.com/pingcap/log@v1.1.1-0.20241212030209-7e3ff8601a2a/global.go:54 +0x85
github.com/pingcap/ticdc/pkg/common/event.SyncPointEvent.encode({0x0, {0x694721831423375a, 0xb12672ebe27cbfb7}, {0xc14602b630, 0x1, 0x1}, 0x2, 0x1, 0x0, {0x0, ...}})
        github.com/pingcap/ticdc/pkg/common/event/sync_point_event.go:135 +0x19b
github.com/pingcap/ticdc/pkg/common/event.SyncPointEvent.Marshal(...)
        github.com/pingcap/ticdc/pkg/common/event/sync_point_event.go:126
github.com/pingcap/ticdc/pkg/messaging.(*remoteMessageTarget).newMessage(0xc142c82900, {0xc0506cf738, 0x1, 0x1?})
        github.com/pingcap/ticdc/pkg/messaging/remote_target.go:662 +0x1f9
github.com/pingcap/ticdc/pkg/messaging.(*remoteMessageTarget).sendEvent(0xc142c82900, {0xc0506cf738, 0x1, 0x1})
        github.com/pingcap/ticdc/pkg/messaging/remote_target.go:123 +0x72
github.com/pingcap/ticdc/pkg/messaging.(*messageCenter).SendEvent(0xc000fd8630, 0xc2c1bfc5b0)
        github.com/pingcap/ticdc/pkg/messaging/message_center.go:303 +0x385
github.com/pingcap/ticdc/pkg/eventservice.(*eventBroker).sendMsg(0xc04f110420, {0x66c44c8, 0xc04f10e6e0}, 0xc2c1bfc5b0, 0x0)
        github.com/pingcap/ticdc/pkg/eventservice/event_broker.go:684 +0xf9
github.com/pingcap/ticdc/pkg/eventservice.(*eventBroker).runSendMessageWorker(0xc04f110420, {0x66c44c8, 0xc04f10e6e0}, 0x14, {0x5b4fc04, 0xf})
        github.com/pingcap/ticdc/pkg/eventservice/event_broker.go:628 +0x656
github.com/pingcap/ticdc/pkg/eventservice.newEventBroker.func1()
        github.com/pingcap/ticdc/pkg/eventservice/event_broker.go:141 +0x33
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.13.0/errgroup/errgroup.go:79 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1631
        golang.org/x/sync@v0.13.0/errgroup/errgroup.go:76 +0x93

### What did you expect to see?

_No response_

### What did you see instead?

Nan

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

--- a/pkg/common/event/sync_point_event.go
+++ b/pkg/common/event/sync_point_event.go
@@ -45,6 +45,16 @@ type SyncPointEvent struct {
 	PostTxnFlushed []func()
 }
 
+func NewSyncPointEvent(id common.DispatcherID, commitTsList []uint64, seq uint64, epoch uint64) *SyncPointEvent {
+	return &SyncPointEvent{
+		DispatcherID: id,
+		CommitTsList: commitTsList,
+		Seq:          seq,
+		Epoch:        epoch,
+		Version:      SyncPointEventVersion,
+	}
+}
+
 func (e *SyncPointEvent) GetType() int {
 	return TypeSyncPointEvent
 }

--- a/pkg/common/event/sync_point_event_test.go
+++ b/pkg/common/event/sync_point_event_test.go
@@ -21,14 +21,8 @@ import (
 )
 
 func TestSyncpointEvent(t *testing.T) {
-	e := SyncPointEvent{
-		State:        EventSenderStatePaused,
-		DispatcherID: common.NewDispatcherID(),
-		CommitTsList: []uint64{100, 102},
-		Seq:          1000,
-		Epoch:        10,
-		Version:      SyncPointEventVersion,
-	}
+	e := NewSyncPointEvent(common.NewDispatcherID(), []uint64{100, 102}, 1000, 10)
+	e.State = EventSenderStatePaused
 	data, err := e.Marshal()
 	require.NoError(t, err)
 	require.Len(t, data, int(e.GetSize()))
@@ -36,18 +30,12 @@ func TestSyncpointEvent(t *testing.T) {
 	var e2 SyncPointEvent
 	err = e2.Unmarshal(data)
 	require.NoError(t, err)
-	require.Equal(t, e, e2)
+	require.Equal(t, *e, e2)
 }
 
 func TestSyncpointEventWithEmtpyCommitTsList(t *testing.T) {
-	e := SyncPointEvent{
-		State:        EventSenderStateNormal,
-		DispatcherID: common.NewDispatcherID(),
-		CommitTsList: []uint64{},
-		Seq:          1000,
-		Epoch:        10,
-		Version:      SyncPointEventVersion,
-	}
+	e := NewSyncPointEvent(common.NewDispatcherID(), []uint64{}, 1000, 10)
+	e.State = EventSenderStateNormal
 	data, err := e.Marshal()
 	require.NoError(t, err)
 	require.Len(t, data, int(e.GetSize()))
@@ -55,5 +43,5 @@ func TestSyncpointEventWithEmtpyCommitTsList(t *testing.T) {
 	var e2 SyncPointEvent
 	err = e2.Unmarshal(data)
 	require.NoError(t, err)
-	require.Equal(t, e, e2)
+	require.Equal(t, *e, e2)
 }

--- a/pkg/eventservice/event_broker.go
+++ b/pkg/eventservice/event_broker.go
@@ -25,7 +25,6 @@ import (
 	"github.com/pingcap/ticdc/pkg/apperror"
 	"github.com/pingcap/ticdc/pkg/common"
 	appcontext "github.com/pingcap/ticdc/pkg/common/context"
-	"github.com/pingcap/ticdc/pkg/common/event"
 	pevent "github.com/pingcap/ticdc/pkg/common/event"
 	"github.com/pingcap/ticdc/pkg/config"
 	"github.com/pingcap/ticdc/pkg/integrity"
@@ -465,7 +464,7 @@ func (c *eventBroker) emitSyncPointEventIfNeeded(ts uint64, d *dispatcherStat, r
 		if len(commitTsList) > 16 {
 			newCommitTsList = commitTsList[:16]
 		}
-		e := event.NewSyncPointEvent(d.id, newCommitTsList, d.seq.Add(1), d.epoch.Load())
+		e := pevent.NewSyncPointEvent(d.id, newCommitTsList, d.seq.Add(1), d.epoch.Load())
 		log.Debug("send syncpoint event to dispatcher", zap.Stringer("dispatcher", d.id), zap.String("eventType", pevent.TypeToString(e.GetType())), zap.Uint64("commitTs", e.GetCommitTs()), zap.Uint64("seq", e.GetSeq()))
 
 		syncPointEvent := newWrapSyncPointEvent(remoteID, e, d.getEventSenderState())

--- a/pkg/eventservice/event_broker.go
+++ b/pkg/eventservice/event_broker.go
@@ -25,6 +25,7 @@ import (
 	"github.com/pingcap/ticdc/pkg/apperror"
 	"github.com/pingcap/ticdc/pkg/common"
 	appcontext "github.com/pingcap/ticdc/pkg/common/context"
+	"github.com/pingcap/ticdc/pkg/common/event"
 	pevent "github.com/pingcap/ticdc/pkg/common/event"
 	"github.com/pingcap/ticdc/pkg/config"
 	"github.com/pingcap/ticdc/pkg/integrity"
@@ -464,12 +465,7 @@ func (c *eventBroker) emitSyncPointEventIfNeeded(ts uint64, d *dispatcherStat, r
 		if len(commitTsList) > 16 {
 			newCommitTsList = commitTsList[:16]
 		}
-		e := &pevent.SyncPointEvent{
-			DispatcherID: d.id,
-			CommitTsList: newCommitTsList,
-			Seq:          d.seq.Add(1),
-			Epoch:        d.epoch.Load(),
-		}
+		e := event.NewSyncPointEvent(d.id, newCommitTsList, d.seq.Add(1), d.epoch.Load())
 		log.Debug("send syncpoint event to dispatcher", zap.Stringer("dispatcher", d.id), zap.String("eventType", pevent.TypeToString(e.GetType())), zap.Uint64("commitTs", e.GetCommitTs()), zap.Uint64("seq", e.GetSeq()))
 
 		syncPointEvent := newWrapSyncPointEvent(remoteID, e, d.getEventSenderState())

